### ISSUE TITLE
Updating multiple NonbondedForces does not work as expected

### ISSUE THREAD

**Issue by JoaoRodrigues** [pandegroup/openmm#2083]
Hi there,

This might be a silly question, but is it impossible to have more than one `NonbondedForce` acting on the system? 

I don't really have a simple piece of code to illustrate my problem, so I'll try to pseudocode it:

```
f1 = mm.NonbondedForce()
f1.setForceGroup(1)
# assign params
f2 = mm.NonbondedForce()
f2.setForceGroup(2)
# assign params

for particle_i in system:
    for particle_j in system:
        f1.setParticleParameters(particle1, x, 0.0, 0.0)
        f1.updateParametersinContext(context)
        ene_1 = context.getState(getEnergy=True, groups={1})

        f2.setParticleParameters(particle2, 0.0, y, z)
        f2.updateParametersinContext(context)
        ene_2 = context.getState(getEnergy=True, groups={2})

        ene_1_again = context.getState(getEnergy=True, groups={1})
```

Running something like this will give `ene_2 == ene_1_again` instead of `ene_1 == ene_1_again`. 

Is my only option implementing `CustomNonbondedForce` objects?

**Comment by jchodera** [comment]
I think that pseudocode *should* work. Is is possible to produce a minimal self-contained test case?

**Comment by JoaoRodrigues** [comment]
As minimal as I get with a realistic system: [link to Gist](https://gist.github.com/JoaoRodrigues/4f772c5705ad072b38952f4a3ec5409e) and [link to PDB](https://gist.github.com/JoaoRodrigues/c7613292c43a8794f8d5647d1f0f0d8a)

The result (first few lines), with `atom_i`, `atom__j`, `ene_1`, `ene_1_again`, and `ene_2`. As you see, it seems `ene_1_again` accumulates `ene_1 + ene_2` although I updated difference forces. Either I am missing something very obvious in the code, or I should just settle for a `CustomNonbondedForce`.

```
1 11 -0.108838537227 -3.5493740844 -3.44053554717
1 12 -0.0543699672812 -3.54920267831 -3.49483275733
1 13 -0.000812143056669 1.21206330068 1.21287542821
1 14 -0.000577917154237 1.00558813854 1.00616606426
1 18 -0.0311427146025 3.32803274888 3.35917554895
1 19 -0.101823943808 -1.64781074907 -1.5459867554
1 20 -0.0136602637901 0.785604033826 0.799264324327
1 21 -0.0515017715455 -0.0543982524204 -0.00289648109744
1 22 -0.00304668357357 0.0684575514287 0.0715042334442
1 23 -0.00652911481634 0.102931024581 0.109460140733
....
```

**Comment by peastman** [comment]
This is the result of an optimization that dates back to the original CUDA code OpenMM is descended from.  It's kind of a hack and I've been a bit unhappy with it for a long time, but removing it would risk hurting performance.  It assumes every atom has a single, well defined charge and sticks it into the fourth slot in the float4 that holds the position.  That means if you have two different NonbondedForces, and if they specify different charges, it doesn't work correctly since they each try to store their charges into the same memory locations.

**Comment by jchodera** [comment]
Would it be possible to only degrade performance if needed by multiple `NonbondedForce`s? It would seem like supporting the API is more important than degrading performance a bit, but it would obviously be great to have the best of both worlds!

**Comment by peastman** [comment]
Yes, that's what I have in find.  It's just going to be a bit of a pain to implement!

**Comment by JoaoRodrigues** [comment]
Ah! I thought it would be something like that. I'll settle for `CustomNonbondedForce` then. Thanks, happy to test this whenever it is pushed!

**Comment by peastman** [comment]
Fixed by #2085.
